internal/pkg/deploy/upload/asset/asset.go

```diff
@@ -174,9 +175,31 @@ func (u *ArtifactBucketUploader) uploadAssetMappingFile(assets []asset) (string,
 		return "", fmt.Errorf("encode uploaded assets: %w", err)
 	}
 
-	path := path.Join(u.AssetMappingFileDir, hex.EncodeToString(hash.Sum(nil)))
-	if err := u.Upload(path, bytes.NewBuffer(data)); err != nil {
+	// include mapping file in hash so any src->dst
+	// mapping changes result in a new hash
+	hash.Write(data)
```

Comment:
I'm trying to enumerate what would happen ⬇️ 💭 which looks good to me!!

Change content of a file ➡️  `ArtifactBucketPath` changes, i.e. the path to which it is uploaded is changed (because the path contains the hash) ➡️ `data` changes ➡️  This `hash` changes, because both `asset.content` and `data` change,

Change `dest` of an upload ➡️ `ServiceBucketPath` changes ➡️  This `hash` changes, because of `data`.

Move a file to a new local path (i.e. change `localPath` without changing `content`) ➡️ `localPath` ➡️  This `hash` does not change

Reply:
based on this ^ i realized we can just hash the asset mapping file and not bother hashing the actual asset file content, since the `ArtifactBucketPath` will only change if the file content changes. updated the pr!